### ISSUE TITLE
How to modify evalTargetFunction() ?

### ISSUE THREAD

**Issue by nbauer2** [mlr-org/mlrMBO#6]
Why is evalTargetFunction() such complex? Can we get rid of "...", because we always get an error here, if we want to evaluate mbo() function step by step.

Regarding the "fun" argument of evalTargetFunction() function: "Fitness function to minimize. The first argument has to be a list of values." Here is unclear whether each parameter has to be a list entry or the whole set, e.g., list(c(22,13)) or list(22,13)?

I would propose to get rid of the list condition and switch to vector. In this case
user can define the objective functions more easy. For example, with the list condition it is impossible to define BBOB functions as follows:
objfun1=generate_ackley_function(dimensions=5) as the first argument is a vector and not a list.

To summarize, at the moment leads the function evalTargetFunction() to the most error messages by trying to apply mbo() function.
